AdminWebsite/AdminWebsite/ClientApp/src/app/booking/summary/summary.component.ts

```diff
@@ -243,72 +240,83 @@ export class SummaryComponent implements OnInit, OnDestroy {
-                        this.logger.info(`${this.loggerPrefix} Hearing has a range of days. Booking remaining days`, {
-                            hearingId: hearingDetailsResponse.id,
-                            caseName: this.hearing.cases[0].name,
-                            caseNumber: this.hearing.cases[0].number
-                        });
-                        await this.hearingService.cloneMultiHearings(
-                            hearingDetailsResponse.id,
-                            new MultiHearingRequest({
-                                start_date: new Date(this.hearing.scheduled_date_time),
-                                end_date: new Date(this.hearing.end_hearing_date_time)
-                            })
-                        );
-                    } else {
-                        this.logger.info(`${this.loggerPrefix} Hearing has just one day, no remaining days to book`, {
-                            hearingId: hearingDetailsResponse.id,
-                            caseName: this.hearing.cases[0].name,
-                            caseNumber: this.hearing.cases[0].number
+                // Poll Video-Api for booking confirmation
+                const schedule = timer(0, 5000).subscribe(async counter => {
+                    const hearingStatusResponse = await this.hearingService.getStatus(hearingDetailsResponse.id);
+                    if (hearingStatusResponse?.success || counter === 10) {
+                        schedule.unsubscribe();
+                        await this.processBooking(hearingDetailsResponse, hearingStatusResponse);
+                        sessionStorage.setItem(this.newHearingSessionKey, hearingDetailsResponse.id);
+                        this.hearingService.cancelRequest();
+                        this.showWaitSaving = false;
+                        this.logger.info(`${this.loggerPrefix} Saved booking. Navigating to confirmation page.`, {
+                            hearingId: hearingDetailsResponse.id
                         });
+                        await this.router.navigate([PageUrls.BookingConfirmation]);
                     }
-                }
-
-                sessionStorage.setItem(this.newHearingSessionKey, hearingDetailsResponse.id);
-                this.hearingService.cancelRequest();
-                this.showWaitSaving = false;
-                this.logger.info(`${this.loggerPrefix} Saved booking. Navigating to confirmation page.`, {
-                    hearingId: hearingDetailsResponse.id
                 });
-                this.router.navigate([PageUrls.BookingConfirmation]);
             } catch (error) {
                 this.logger.error(`${this.loggerPrefix} Failed to save booking.`, error, { payload: this.hearing });
                 this.setError(error);
             }
         }
     }
 
+    async processBooking(hearingDetailsResponse, hearingStatusResponse): Promise<void> {
+        if (hearingStatusResponse?.success) {
```

Comment:
From the code coverage results it looks like we might be missing coverage for when this success response is true. Perhaps we are missing a test?

Reply:
added tests